BUILDING.md

```diff
@@ -110,17 +125,28 @@ platforms. This is true regardless of entries in the table below.
 
 Depending on the host platform, the selection of toolchains may vary.
 
-#### Unix
-
-* GCC 4.9.4 or newer
-* Clang 3.4.2 or newer
-
-#### AIX
-* GCC 6.3 or newer
-
-#### Windows
-
-* Visual Studio 2017 with the Windows 10 SDK on a 64-bit host.
+| Operating System | Compiler Versions                                           |
+| ---------------- | ----------------------------------------------------------- |
+| Linux            | GCC >=6.3, Clang >=3.4.2                                    |
+| Windows          | Visual Studio 2017 with the Windows 10 SDK on a 64-bit host |
+| macOS            | Xcode / Apple LLVM >= 7                                     |
```

Comment:
TODO: pending discussion

Reply:
For Windows should be MSVS 2019 which is in RC already, and will be released before node12 on April 2nd.
For macOS IMHO we should state XCode level (not LLVM), since that is how we test.

Reply:
Windows: I've added a `>=` for MSVS to suggest we support later too. As @joaocgreis pointed out in our email discussion, we have had problems in the past with jumping on brand new versions, I think we ran into this with 2015 and the bulk of the support burden fell on @joaocgreis and his colleagues. I'm inclined to agree with @joaocgreis about not being so aggressive with MSVS versions.

Reply:
Mac: the reason I didn't is because I couldn't figure it out from our test machines! When you have only the Xcode command line tools installed it's very difficult to discern the Xcode version other than by the compiler version. I've added "Xcode 7" in now but I'm hoping someone's going to show up with a solid recommendation on this.

Reply:
To be fair, @refack is usually a huge help in this type of issues.

I'd much rather we keep the minimum at VS2017 than jumping into VS2019 exclusively when it won't even be one month old. We might be forced to drop VS2017 for some reason during Node 12 lifetime, but I see that as less likely to give us problems than jumping into a brand new compiler right away.

Reply:
Learned this for my GYP3 work:
```
pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
```

Reply:
Went back down to Xcode 9 as per the discussion on 10.12 above, and I've listed it here as "Xcode >= 9 (Apple LLVM >= 9)"

Reply:
Leaving Windows as is, at least that's my interpretation of the conversation here, "Visual Studio >= 2017 with the Windows 10 SDK on a 64-bit host"